integrationtest/initial_release/main.d

```diff
@@ -0,0 +1,182 @@
+/*******************************************************************************
+
+    Tests the 'first release' scenario with the neptune-release tool
+
+    Copyright:
+        Copyright (c) 2017 Sociomantic Labs GmbH. All rights reserved.
+
+    License:
+        Boost Software License Version 1.0. See LICENSE.txt for details.
+
+*******************************************************************************/
+
+module integrationtest.initial_release.main;
+
+import integrationtest.initial_release.GHTestServer;
+import vibe.web.rest;
+import vibe.http.router;
+
+/// Set to true if the test failed
+bool failed;
+
+/*******************************************************************************
+
+    Main function, sets up tests & runs event loop
+
+*******************************************************************************/
+
+version(UnitTest) {} else
+void main ( )
+{
+    import vibe.core.core;
+    import std.stdio;
+
+    RestAPI gh_test;
+
+
+    auto router = new URLRouter;
+	router.registerRestInterface(gh_test = new RestAPI);
+
+	auto settings = new HTTPServerSettings;
+	settings.port = 8080;
+	listenHTTP(settings, router);
+
+    import std.functional : toDelegate;
+
+    auto task = runTask(toDelegate(&runNeptune), gh_test);
+
+    runEventLoop();
+
+    task.join();
+
+    assert(!failed);
+}
+
+/*******************************************************************************
+
+    Runs the test
+
+    Params:
+        gh_test = github mockup server instance
+
+*******************************************************************************/
+
+void runNeptune ( RestAPI gh_test )
+{
+    import integrationtest.initial_release.shellHelper;
+
+    import vibe.core.core;
+    import vibe.core.args;
+
+    import std.stdio;
+    import std.process;
+    import std.format;
+
+    import core.time;
+
+    enum GitRepo = "tester/sandbox";
+    enum GitPath = "github.com:" ~ GitRepo;
+
+    scope(failure)
+        failed = true;
+
+    scope(exit)
+        exitEventLoop();
+
+    auto tmp = readRequiredOption!string("tmp", "Temporary directory");
+    auto bin = readRequiredOption!string("bin", "Binary directory");
+    auto data = __FILE_FULL_PATH__[0 .. $-"main.d".length] ~ "data";
```

Comment:
Ha ha, so one can't move project folder without all tests starting to fail? :)

Reply:
I don't understand? The path is always the current file, so moving it also updates the variable... of course you need to recompile, yes.